crates/xilem_core/src/message.rs

```diff
@@ -3,21 +3,60 @@
 
 use std::any::Any;
 
-use crate::id::IdPath;
+#[macro_export]
```

Comment:
Why does this need to be generic? My understanding was that `Message` and related types do not really depend on the types of the element tree (in particular, they shouldn't need a bound on the trait). But it's entirely possible I'm missing something.